docs/developers-guide.md

```
@@ -1268,6 +1380,140 @@ assert_eq!(
+so local setup instructions stay truthful.
+
+- Issue `#35` (PR `#168`): `DefaultSelfRepair::repair_broken_tool` was
+  refactored to ≤ 35 lines by extracting `build_repair_requirement`,
+  `handle_build_result`, and `attempt_repair_build` as private static
+  helpers. No functional changes were made. See
+  `src/agent/self_repair/default.rs` and
+  `src/agent/self_repair/default_tests/`.
+
+## 33. Phased startup pipeline
+
+## 32. Expected follow-up changes
```

markdownlint-cli2 (0.22.1)

[warning] Multiple headings with the same content

(MD024, no-duplicate-heading)